pkg/zone/feature/meter/group.go

```
@@ -77,6 +80,23 @@ func (g *Group) GetMeterReading(ctx context.Context, request *gen.GetMeterReadin
 		i := i
 		fns[i] = func() {
 			res, err := g.apiClient.GetMeterReading(ctx, request)
+			if err != nil && g.useHistoryBackupOnErr {
+				// try to get the latest backup reading from the history table if the main read fails
+				latest, historyErr := g.historyApiClient.ListMeterReadingHistory(ctx, &gen.ListMeterReadingHistoryRequest{
```

I think at the very least, we should only use data if it is within a certain time frame. Using a reading from a few hours / a day or so ago for a zone approximation I think is fine. But if the latest history reading is older than some value, then we throw the error. Also, if more than x% of meters are throwing errors getting data, then we should raise the error. 

Mitigating temporary issues with a few meters is not a bad idea IMO, however there has to be some point where we just have to say that we can't show the zone/building data as there is a major issue with the underlying data. This error needs to be surfaced on the graph though, just having an empty graph with no data and no explanation of why the data isn't there is bad.

---

Yes more just me thinking and typing for the UI stuff, have created [SCB-1231](https://vanti.youtrack.cloud/issue/SCB-1231) Show error message on energy graph when no data is available